### ISSUE TITLE
Add line number reporting to NoUnusedImports

### ISSUE THREAD

**Issue by matejdro** [detekt/detekt#6560]
## Expected Behavior of the rule

For example, for the following file:

```kotlin
package mypackage

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flowOf
import kotlinx.coroutines.flow.onCompletion
import kotlinx.coroutines.flow.onStart

fun main() {
    flowOf("").onStart {}
}
```

`NoUnusedImports` rule will just report beginning of file for all imports:

```
NoUnusedImports - [Unused import] at Test.kt:1:1
NoUnusedImports - [Unused import] at Test.kt:1:1
``` 

## Context

This can make it fairly annoying to determine which imports exactly are unused (without the use of IDE). It also makes this rule inferior to the duplicate ktlint's `UnusedImports` rule, which reports line numbers fine.

**Comment by cortinico** [comment]
> It also makes this rule inferior to the duplicate ktlint's UnusedImports rule

The `NoUnusedImports` is from ktlint.

The rule `UnusedImport` instead is from Detekt:
https://detekt.dev/docs/next/rules/style/#unusedimport

**Comment by matejdro** [comment]
Sorry, I mixed the two up. Closing.
